src/main/javascript/crypto/src/eas/AttestationUrl.ts

```diff
@@ -0,0 +1,312 @@
+
+import * as pako from "pako";
+import * as Base64 from "js-base64";
+import {SignedOffchainAttestation} from "@ethereum-attestation-service/eas-sdk";
+import {ethers} from "ethers";
+
+export interface SignedOffchainAttestationV1
+	extends Omit<SignedOffchainAttestation, "signature"> {
+	r: string;
+	s: string;
+	v: number;
+}
+
+export interface AttestationShareablePackageObject {
+	/** Signed typed data with attestation object */
+	sig: SignedOffchainAttestation | SignedOffchainAttestationV1;
+	/** Address of the signer */
+	signer: string;
+}
+
+export type CompactAttestationShareablePackageObject = [
+	version: string,
+	chainId: number,
+	verifyingContract: string,
+	r: string,
+	s: string,
+	v: number,
+	signer: string,
+	uid: string,
+	schema: string,
+	recipient: string,
+	time: number,
+	expirationTime: number,
+	refUID: string,
+	revocable: boolean,
+	data: string,
+	nonce: number
+];
+
+export type CompactLightweightAttestationShareablePackageObject = [
+	r: string,
+	s: string,
+	v: number,
+	uid: string,
+	recipient: string,
+	time: number,
+	expirationTime: number,
+	refUID: string,
+	revocable: boolean,
+	data: string,
+	nonce: number
+];
+
+export interface StaticSchemaInformation {
+	domain: {
+		version: string,
+		chainId: number,
+		verifyingContract: string,
+	},
+	schema: string,
+	signer: string,
+}
+
+export function createURL(attPackage: AttestationShareablePackageObject) {
+	const base64 = zipAndEncodeToBase64(attPackage);
+	return `#attestation=${encodeURIComponent(base64)}`;
+}
+
+export function zipAndEncodeToBase64(
+	qrPackage: AttestationShareablePackageObject,
+	lightweight = false
+) {
+	const compacted = lightweight ?
+		compactLightweightOffchainAttestationPackage(qrPackage) :
+		compactOffchainAttestationPackage(qrPackage);
+
+	const jsoned = JSON.stringify(compacted);
+
+	const gzipped = pako.deflate(jsoned, { level: 9 });
+	return Base64.fromUint8Array(gzipped);
+}
+
+export function decodeBase64ZippedBase64(
+	base64: string,
+	schemaInfo?: StaticSchemaInformation
+): AttestationShareablePackageObject {
+	const fromBase64 = Base64.toUint8Array(base64);
+
+	const jsonStr = pako.inflate(fromBase64, { to: "string" });
+
+	const compacted: CompactAttestationShareablePackageObject|CompactLightweightAttestationShareablePackageObject =
+		JSON.parse(jsonStr);
+
+	return schemaInfo ?
+		uncompactLightweightOffchainAttestationPackage(compacted as CompactLightweightAttestationShareablePackageObject, schemaInfo) :
+		uncompactOffchainAttestationPackage(compacted as CompactAttestationShareablePackageObject);
+}
+
+export function compactLightweightOffchainAttestationPackage(
+	pkg: AttestationShareablePackageObject
+): CompactLightweightAttestationShareablePackageObject {
+	let { sig, signer } = pkg;
+
+	if (isSignedOffchainAttestationV1(sig)) {
+		sig = convertV1AttestationToV2(sig);
+	}
+
+	return [
+		sig.signature.r,
+		sig.signature.s,
+		sig.signature.v,
```

Comment:
@oleggrib good points, but we should aim to maintain compatibility with whatever EAS does. Steve said he would be including this in the SDK in the future. He is also looking into the improvements I made by using binary encoding instead of JSON, so maybe that will be a part of the SDK too :-)